### PR TITLE
feat(sdk): Handle subdirectories for the folder in S3Validator

### LINKED ISSUE #4100: Update S3 Validator to handle subdirectories for the folder
Our S3Validator class assumes that the S3 URI will have single directory where validators will save checkpoints. Some validators are saving checkpoints in subdirectories.

Our agents are able to handle subdirectories and can read the checkpoints. We want to update our S3Validator class to be aligned with the agents and work in the same way

### PR DESCRIPTION
**Pull request by Mo-Hussain**
### Description

- Set the folder in the `S3Config` to include any subdirectories and not just the first folder
- This brings the SDK inline with current behavior in the agents

### Related issues

- Fixes #4100 

### Backward compatibility

Yes

### Testing

Manual

### CODE REVIEW

File: typescript/sdk/src/aws/validator.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export class S3Validator extends BaseValidator {
         const s3Config = {
           bucket: pieces[0],
           region: pieces[1],
-          folder: pieces[2],
+          folder: pieces.slice(2).join('/'),
```
Comment (paulbalaji) [inline typescript/sdk/src/aws/validator.ts:42]: might be my smooth brain but can you give me an example of before/after this fix pls

**Review by jmrossy — APPROVED** [review]
lgtm

File: typescript/sdk/src/aws/validator.ts (hunk 1 above)
Comment (Mo-Hussain) [reply typescript/sdk/src/aws/validator.ts:42]: The existing implementation: Coinsummer are storing the objects in a sub directory, `s3://cs-avs-hyperlane/us-east-1/validator/avs1/ethereum`, we are looking in `validator` and not finding anything.

new implementation will look in `validator/avs1/ethereum`

**Review by daniel-savu — APPROVED** [review]
nice, just ran and it we have 30 validators running correctly according to the script (including coinsummer) 🔥

**Pull request merged**